modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/netty/listener/HttpRequestWorker.java

```diff
@@ -338,7 +338,12 @@ private void sendResponse(int statusCode, HttpResponseStatus responseStatus, boo
 
         HttpCarbonMessage outboundResponse;
         try {
-            outboundResponse = new HttpCarbonMessage(new DefaultHttpResponse(HttpVersion.HTTP_1_1, responseStatus));
+            HttpVersion version = HttpVersion.HTTP_1_1;
+            if (sourceConfiguration.getProtocol().equals(BridgeConstants.HTTP_2_0_VERSION)) {
```

Comment:
Shall we call the equals() method on BridgeConstants.HTTP_2_0_VERSION instead of on sourceConfiguration.getProtocol() to avoid any NPEs?

Reply:
changed it